packages/fileeditor-extension/src/commands.ts

```diff
@@ -567,7 +567,11 @@ export namespace Commands {
         if (name && widget) {
           const spec = languages.findByName(name);
           if (spec) {
-            widget.content.model.mimeType = spec.mime as string;
+            if (Array.isArray(spec.mime)) {
+              widget.content.model.mimeType = spec.mime[0] as string;
+            } else {
+              widget.content.model.mimeType = spec.mime as string;
```

Comment:
Is the casting here needed? (genuinely asking)

```suggestion
              widget.content.model.mimeType = spec.mime;
```

If it is not an array it should be a `string` as per:

https://github.com/jupyterlab/jupyterlab/blob/bd46c76046671ae375b3ddd207ca441841e57123/packages/codemirror/src/token.ts#L270

Reply:
while this might be unnecessary, vscode does show a TS2322 error without the casting.
i would not want that distractive red error line in there :)

Reply:
Right, I think that in typescript marking it as `const` ahead would make the compiler infer types correctly:

```ts
const mime = spec.mime;
if (Array.isArray(mime)) {
   // `mime` is now inferred to be an array
} else {
    // `mime` is now inferred to be a string
}
```

see [here](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJIBUIGczIN4CwAUMqcgLbDkQBcyOUoA5sgD71iMhMDaAugG5iAX2IIA9iBzJgyALz5hyOFjSYcQomMnTK1eTIB0eiMWAwAFAEEoUOAE9DwLDbv2LJgJSf8xMsgkpXCgIADcIOAAbdHsAB1oOLl4+AxNNJQhIrBRCEjJA6RDwqJj4ugZmVKoIdKA)

Reply:
i see, but we only want strings here for mimeType so i guess leaving the definition at 'mimeType: string;' feels like the right thing to do?

Reply:
Just checking if you saw my response here @jans-code

Reply:
now i have :)

the if case does work without casting the string (also without a const declaration beforehand), but the else still shows the ts2322.

so this would be as tidy as i can get it:

```typescript
if (Array.isArray(spec.mime)) {
              widget.content.model.mimeType = spec.mime[0] ?? IEditorMimeTypeService.defaultMimeType;
            } else {
              widget.content.model.mimeType = spec.mime as string;
            }
```